### ISSUE TITLE
Provide Gradle Plugin to scan project

### ISSUE THREAD

**Issue by lostiniceland** [jQAssistant/jqassistant#402]
Currently there is only a custom Gradle plugin to scan a project with JQAssistant. The configuration is much more complex compared to the official maven-plugin.

http://techblog.kontext-e.de/jqassistant-with-gradle/

It would be great if there would be an official Gradle plugin

**Comment by kontext-e** [comment]
Hi,
are you looking for this: https://plugins.gradle.org/plugin/de.kontext_e.jqassistant.gradle?
At least it makes the integration in the build process easier. Not yet included is the functionality of the Maven plugin to use intrinsic Gradle dependencies knowledge to improve the class scanning. 

Jens

**Comment by lostiniceland** [comment]
Yes, I was trying this one. First thanks for the effort, but I havent been able to successfully adopt it to a Gradle Kotlin DSL.
For instance I wasnt sure if the additional tasks are needed or if they are just explaining what the plugin creates.

Hence, this issue to get something "official". Maybe your solution could be a valid starting point

**Comment by DirkMahler** [comment]
We would be very happy to provide an official integration with Gradle. The current problem is simply lack of expertise with it, any support highly welcome, we would do our best to communicate/adopt the required integration points from the jQA side!

**Comment by DirkMahler** [comment]
Coming from Maven the following aspects would be important:

* Integration into the build lifecycle, i.e. for starting/stopping the integrated Neo4j database
* Configuration (e.g. database, rules)
* Scanning modules, including their dependencies
* Module/project meta-information that might be relevant for analysis or validation (e.g. build configuration, properties)
